src/jit/codegenxarch.cpp

```diff
@@ -7890,7 +7890,90 @@ CodeGen::getBaseVarForPutArgStk(GenTreePtr treeNode)
     {
         // The first baseVarNum is guaranteed to be the first incoming arg of the method being compiled.
         // See lvaInitTypeRef() for the order in which lvaTable entries are initialized.
+        // For Windows the value is always 0 (there are always slots in the caller's frame for the first 4 args to a callee).
+        // For System V the first stack arg should be calculated.
```

Comment:
It is not clear from this comment and the function header comment that this method is really intended to compute the varNum of the first stack argument location. Please add notes to the function header to clarify what this method is really doing, and what it is used for.